crates/ruff_linter/src/rules/refurb/rules/subclass_builtin.rs

```diff
@@ -70,11 +70,18 @@ pub(crate) fn subclass_builtin(checker: &Checker, class: &StmtClassDef) {
         return;
     };
 
+    // Expect only one base class else return
     let [base] = &**bases else {
         return;
     };
 
-    let Some(symbol) = checker.semantic().resolve_builtin_symbol(base) else {
+    // Check if the base class is a subscript expression so that only the name expr
+    // is checked and modified.
+    let base_expr = base
+        .as_subscript_expr()
+        .map_or(base, |subscript| &*subscript.value);
```

Comment:
I think we could use `map_subscript` here, similar to e.g. 

https://github.com/astral-sh/ruff/blob/46fe17767d3d5cf7b1f7729b024fca7821d19264/crates/ruff_linter/src/rules/flake8_bugbear/rules/mutable_contextvar_default.rs#L98-L100

Reply:
didn't know about the `map_subscript` function. Just modified the logic to use it, seems a lot more clean.